bin/protoc-gen-ruby_gapic_dump

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env ruby
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+$LOAD_PATH << ::File.expand_path('../../lib', __FILE__)
+require 'google/gapic/schema'
+require 'google/gapic/generator'
+require 'protobuf/descriptors'
+
+# Ensure that no encoding conversions are done on STDIN and STDOUT since
+# we are passing binary data back and forth. Otherwise these streams
+# will be mangled on Windows.
+STDIN.binmode
+STDOUT.binmode
+
+# Get the CodeGeneratorRequest.
+request_bytes = STDIN.read
+request = Google::Protobuf::Compiler::CodeGeneratorRequest.decode(request_bytes)
+
+# Create an API Schema from the FileDescriptorProtos
+api = Google::Gapic::Schema::Api.new(request.proto_file, "")
+
+# Specify where to load the templates from.
+class DumpTemplateProvider < Google::Gapic::Generator::TemplateProvider
+  DUMP_TEMPLATE = %{
+<%=@util.file_name "gapic_dump.txt"%>
+<% @api.messages.each do |m| %>
+Message: <%= m.name %>
+  Leading Comments: <%= m.leading_comments.inspect%>
```

Comment:
Just a style nitpick, but the spacing inside the template delimiters appears inconsistent.

Reply:
I made them consistent. I wonder if there is a style checker for erb files.